File: src/DayTwo/Part2.java
package DayTwo;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Part2 {

  public static void main(String[] args) throws FileNotFoundException {
    List<String> inputs = new ArrayList<>();
    File file = new File("inputs/day-two-input.txt");
    Scanner sc = new Scanner(file);

    int horiz = 0;
    int depth = 0;
    int aim = 0;

    while (sc.hasNext()) {
      switch (sc.next()) {
        case "forward":
          int x = sc.nextInt();
          horiz += x;
          depth += aim * x;
          break;
        case "up":
          aim -= sc.nextInt();
          break;
        case "down":
          aim += sc.nextInt();
        default:
          break;
      }
    }
    System.out.println("Final horizontal position: " + horiz);
    System.out.println("Final depth: " + depth);
    System.out.println("Final aim: " + aim);
    System.out.println("Final result (Multiplied): " + horiz * depth);
  }
}

File: python/DayTwo/Part2.py
def read_inputs():
    inputs = []
    with open("inputs/day-two-input.txt", "r") as file:
        for line in file:
            single_line = line.strip()
            inputs.append(single_line)
    return inputs


def process(inputs):
    horiz = 0
    depth = 0
    aim = 0

    for i in range(len(inputs)):
        split = inputs[i].split()
        match split[0]:
            case "forward":
                horiz += int(split[1])
                depth += aim * int(split[1])
            case "up":
                aim -= int(split[1])
            case "down":
                aim += int(split[1])
            case _:
                print("Error; forward, up, or down not parsed")
    print("Final horizontal position: ", horiz)
    print("Final depth: ", depth)
    print("Final aim: ", aim)
    print("Final result (Multiplied): ", horiz * depth)


def main():
    process(read_inputs())


if __name__ == "__main__":
    main()
